### PR TITLE
WIP: Added Armtronix dimmer support

### PR DESCRIPTION
**Pull request by wvdv2002**
There is an interesting Dual AC Dimmer [design](https://www.tindie.com/products/Armtronix/wifi-ac-dimmer-two-triac-board/) from Armtronix which I decided to add support to Tasmota for, as the [original firmware](https://github.com/armtronix/Wifi-Two-Dimmer-Board)  is not optimal. The hardware design is though.

Thanks to @armtronix for the dimmer hardware design.

The dimmer contains an ESP8266 for the wifi connection and an Atmega328 for controlling the dimming.
Be sure to update the [software](https://github.com/wvdv2002/Wifi-Two-Dimmer-Board/tree/master/Arduino_Code/Atmega328_two_dimmerV0.4_Tasmota) in the Atmega328 as well.

I added an extra module " ARMTRONIX_DIMMERS" for use with this dimmer.
The next part will be to change the software of the single channel [dimmer](https://www.tindie.com/products/Armtronix/wifi-ac-dimmer-esp8266-one-triac-board-alexaecho/) so it can be used with Tasmota as well.

***** EDIT***************
I have added [code](https://github.com/wvdv2002/Wifi-Single-Dimmer-Board) for the single[dimmer](https://www.tindie.com/products/Armtronix/wifi-ac-dimmer-esp8266-one-triac-board-alexaecho/) now as well. They will now both work with the same Tasmota driver. I made the serial communication the same for the dual and single dimmer version.

### CODE REVIEW

File: sonoff/language/en-GB.h [hunk 1]
```diff
@@ -529,6 +529,10 @@
 #define D_SENSOR_RFRECV "RFrecv"
 #define D_SENSOR_TUYA_TX "Tuya Tx"
 #define D_SENSOR_TUYA_RX "Tuya Rx"
+#define D_SENSOR_ARMTRONIX_TX "Armtronix Tx" 
+#define D_SENSOR_ARMTRONIX_RX "Armtronix Rx"
+#define D_SENSOR_ARMTRONIX_TX "Armtronix Tx"
```
Comment (ascillato) [inline sonoff/language/en-GB.h]: why it is added twice?

File: sonoff/sonoff_template.h [hunk 2]
```diff
@@ -1217,6 +1225,22 @@ const mytmplt kModules[MAXMODULE] PROGMEM = {
...
+     GPIO_USER,
+     GPIO_USER,        // GPIO03 MCU serial control
+     GPIO_USER,
+     GPIO_USER,
+     0, 0, 0, 0, 0, 0, // Flash connection
+     GPIO_USER,
+     GPIO_USER,
+     GPIO_USER,        // GPIO14 Green Led
+     GPIO_USER,
+     GPIO_USER,
```
Comment (ascillato) [inline sonoff/sonoff_template.h]: all GPIO_USER? no key, no buttons, no leds, no relays, no pwm? So why not use Generic instead?

**Review by ascillato — CHANGES_REQUESTED** [review]

File: sonoff/sonoff_template.h [hunk 3]
```diff
@@ -1217,6 +1225,22 @@ const mytmplt kModules[MAXMODULE] PROGMEM = {
      GPIO_USER,
      GPIO_USER,
      0
+  },
+    { "ARMTR Dimmr",     // ARMTRONIX Dimmer (ESP8266 w/ separate MCU dimmer)
+                       // https://www.amazon.com/gp/product/B07CTNSZZ8/ref=oh_aui_detailpage_o00_s00?ie=UTF8&psc=1
```
Comment (ascillato) [inline sonoff/sonoff_template.h]: that link is not the hardware you want to add.

**Review by ascillato — CHANGES_REQUESTED** [review]

**Review by ascillato2 — CHANGES_REQUESTED** [review]
Please address the requested changes. Thanks.

File: sonoff/sonoff_template.h (hunk 2 above)
Comment (wvdv2002) [reply sonoff/sonoff_template.h]: Well, it is a new kind of module in the most physical sense. There are some buttons, I will try to find out where they are connected (although they are not accesible outside of the module and while powered by 230VAC you probably don't want to touch them anyway...)

When I now think about it, my recommendation would be to keep the extra module, as the hardware is physically different. But to not clutter everything up too much, remove the new RX and TX defines. Will propose another solution in an updated pull request.

**Comment by ascillato** [comment]
> Will propose another solution in an updated pull request by using GPIO_TXD and GPIO_RXD instead of newly defined GPIO_ARMTRONIX_RX and GPIO_ARMTRONIX_TX

Seems to be a good option.

**Comment by ascillato** [comment]
Ant news on this?

**Comment by wvdv2002** [comment]
I rewrote parts of the code. My code was copied from the Tuya dimmer code and now that I understand the Tasmota code better I knew that I could remove certain parts. If there are only issues left let me know. Otherwise please merge this pull request.

**Comment by wvdv2002** [comment]
Thanks for accepting and merging.

**Comment by thxthx0** [comment]
Thx @wvdv2002, @ascillato and @arendst for all the effort,

but how does a TRIAC-dimmer work without big inductors and caps,
with all the spikes?

and what's the purpose of a TRIAC-dimmer these days, 
beside tungsten lamps?

**Comment by wvdv2002** [comment]
Hi thxthx0,

Well, this is mostly an issue with inductive loads, not with resistive loads.
I made a note of your worries in the warnings section of the dimmer's [wiki](https://github.com/arendst/Sonoff-Tasmota/wiki/Armtronix-Dimmers). 

There are quite a few dimmable led GU10 'halogen replacement' spots. As noted in the Wiki, IKEA for example has quite a few good ones. And the nice thing is that you can use your current light fixtures and wiring.

**Pull request merged**